gravitee-node-api/src/main/java/io/gravitee/node/api/upgrader/Upgrader.java

```diff
@@ -21,6 +21,17 @@
  * @author GraviteeSource Team
  */
 public interface Upgrader {
+    /**
+     * Implementations can override this method to state that they ran without
+     * committing any change. If the return is true, the upgrade will not be stored
+     * and will run again on subsequent executions until the implementation returns false.
+     *
+     * @return true if the upgrader actually committed its changes, false otherwise.
+     */
+    default boolean isDryRun() {
```

Comment:
I am not sure to understand the feature. From the javadoc here, I understand the changes are not committed and no changes are made in the repository. From the code, I understand that only the UpgradeRecord is not committed to the database but any changes made by the upgrader are still performed. Could you explain ?

Reply:
All the changes made by the upgrader are in the dry run mode.
So it means nothing has been committed to the database by the upgrader actually.

Reply:
Is it up to the upgrader to manage that commitment?

Reply:
Yes, at least at the moment it is implemented like this. It can be improved later on. The concept on dry run is used only in APIM, btw.